Fix Cooldown for Raise Dead (Ghoul)

tkrokli opened the pull request:
Fix written by @joschiwald on 31 Jan 2014
Added null pointer check suggested by @ariel-
Fixes issue #11538

Commit:
Fix Cooldown for Raise Dead (Ghoul)

Fix written by joschiwald on 31 Jan 2014
Added null pointer check suggested by @ariel-
Fixes issue #11538

ccrs commented on src/server/game/Entities/Player/Player.cpp:
```diff
@@ -22138,6 +22138,35 @@ void Player::SendCooldownEvent(SpellInfo const* spellInfo, uint32 itemId /*= 0*/
+    
+    uint32 cat = spellInfo->GetCategory();
+    uint32 catRec = spellInfo->CategoryRecoveryTime;
+    if (cat && catRec)
```
catRec is never used inside, maybe change to
````c++
if (cat && spellInfo->CategoryRecoveryTime)
````

tkrokli replied:
OK, but I would like to have that confirmed before I start changing the code again.

ccrs commented on src/server/game/Entities/Player/Player.cpp:
```diff
@@ -22138,6 +22138,35 @@ void Player::SendCooldownEvent(SpellInfo const* spellInfo, uint32 itemId /*= 0*/
+    if (cat && catRec)
+    {
+        SpellCategoryStore::const_iterator ct = sSpellsByCategoryStore.find(cat);
+        if (ct != sSpellsByCategoryStore.end())
```
Dont know how it should be done, but it looks pretty ugly.

If i got it right, the code searches all spells in dbc with same category for every spell used by a player.
Then inside every category it searches if any of those spells has a "cooldown" stored in  the player.

Imagine how many times this double loop can be executed...

tkrokli replied:
Thanks for pointing that out. I wish I knew how to deal with that.

ccrs commented:
I know that something is better than anything, but surely a developer can find a better solution for this.

Any thoughts? @joschiwald

tkrokli commented:
@Aokromes : would you mind adding the same labels **[Branch-3.3.5a]** + **[Feedback-PatchFix]** as in my previous PR, #13924 ? Thanks in advance. :+1:

Commit:
Fix Cooldown for Raise Dead (Ghoul)

Fix written by joschiwald on 31 Jan 2014. Null pointer check suggested by @Nyeriah. Actual code for null pointer check supplied by @ariel- . Replace (cat && catRec) with (cat && spellInfo->CategoryRecoveryTime) provided by @ccrs.

Fixes issue #11538

tkrokli commented:
Hmm, I am unable to make the guide in https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#squash-your-changes-optional work for me, I always get it wrong in git bash when I get to the point `git merge-base my-branch master` because I can't find out what the keywords `my-branch` and `master` should be replaced by. It ends up looking like: `git merge-base 3.3.5 3.3.5` and it does not seem to fit, since I have no idea which one of them is remote and which one of them is local.

The next step does not produce any pick list, only "noop" which I guess is a message that there is nothing to do or the command was wrong (returning "nope").

For the time being, until I understand how to run git bash (which is more complicated than compiling my own custom Linux kernel), I have to delete my fork and create a new fork and base my PR on that new fork (hence closing this PR and creating yet another PR). There just is not any compatibility between working directly on my own fork on the web page and the current TC way of cloning, pushing and pulling. Sorry about that. I'm just too slow in my brain.

tkrokli commented:
Closing PR in favor of the updated #13963

tkrokli commented:
#13948 - closed as a duplicate, but pointing to this PR before closing.

DDuarte commented:
You never ever have to delete and recreate the fork to remove changes. Without git checkout and git push --force it's not needed. You simply rollback to an older version and force push, new changes will be gone. (Don't forget git pull --rebase to get the new changes).

The guide assumes that each PR has a different branch (because that's how it should be done, never push directly to master (in this case 3.3.5)).
There are other guides to squash, try https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/.
